File: RepositoryNotifier/DTO/JobTO.cs
using System;
using System.Collections.Generic;
using RepositoryNotifier.Persistence.Job;

namespace RepositoryNotifier.DTO
{
    public class CreateJobTO
    {
        public string Username { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public IList<string> Repositories { get; set; }
        public JobFrequency Frequency { get; set; }
        public IList<string> SearchKeywords{get;set;}
        public bool EmailNotificationEnabled { get; set; }
        public bool SmsNotificationEnabled { get; set; }
        public bool WhatsappNotificationEnabled { get; set; }
        public bool SchedulerEnabled {get;set;}

    }


    public class JobResultTO{
        public string Name {get;set;}
        public string Path {get;set;}
        public string Url {get;set;}
        public string RepositoryName {get;set;}
        public DateTime CreatedAt {get;set;}
    }

}

File: RepositoryObserver/ClientApp/src/app/dto/jobTO.ts
export class AddJobTO{
    username: string;
    email: string;
    repositories: string[];
    frequency: number;
    searchKeywords: string[];
    phoneNumber:string;
    emailNotificationEnabled:boolean;
    smsNotificationEnabled:boolean;
    whatsappNotificationEnabled:boolean;
    schedulerEnabled:boolean;
}

export class UpdateJobTO{
    username: string;
    email: string;
    repositories: string[];
    frequency: number;
    searchKeywords: string[];
}

export class GetJobTO extends AddJobTO{
    id: number;
    createdAt: Date;
    updatedAt: Date;
    lastExecutedAt: Date;
    status: string;
}

export class Job{
    getJobTO: GetJobTO;
    selected: boolean;


    constructor(getJobTO: GetJobTO, selected: boolean) {
        this.getJobTO = getJobTO;
        this.selected = selected;
    }
}

export class DeleteJobTO{
    email: string;
    frequency: number;
}

export class JobResultTO{
    name:string;
    path:string;
    url:string;
    repositoryName:string;
    createdAt: Date;
}
